src/PostgREST/Response.hs

```diff
@@ -152,27 +154,28 @@ actionResponse (DbCrudResult MutateReadPlan{mrMutation=MutationUpdate, mrMedia}
     Right $ PgrstResponse ovStatus ovHeaders body
 
   RSPlan plan ->
-    Right $ PgrstResponse HTTP.status200 (contentTypeHeaders mrMedia ctxApiRequest) $ LBS.fromStrict plan
+    Right $ PgrstResponse HTTP.status200 (contentLengthHeader plan : contentTypeHeaders mrMedia ctxApiRequest) $ LBS.fromStrict plan
 
 actionResponse (DbCrudResult MutateReadPlan{mrMutation=MutationSingleUpsert, mrMedia} resultSet) ctxApiRequest@ApiRequest{iPreferences=Preferences{..}} _ _ _ _ _ = case resultSet of
   RSStandard {..} -> do
     let
       prefHeader = maybeToList . prefAppliedHeader $ Preferences Nothing preferRepresentation preferCount preferTransaction Nothing preferHandling preferTimezone Nothing []
+      cLHeader = [contentLengthHeader rsBody]
       cTHeader = contentTypeHeaders mrMedia ctxApiRequest
 
     let isInsertIfGTZero i = if i > 0 then HTTP.status201 else HTTP.status200
         upsertStatus       = isInsertIfGTZero $ fromJust rsInserted
         (status, headers, body) =
           case preferRepresentation of
-            Just Full -> (upsertStatus, cTHeader ++ prefHeader, LBS.fromStrict rsBody)
+            Just Full -> (upsertStatus, cLHeader ++ cTHeader ++ prefHeader, LBS.fromStrict rsBody)
             Just None -> (HTTP.status204,  prefHeader, mempty)
```

Comment:
[According to the spec](https://httpwg.org/specs/rfc9110.html#field.content-length), the `204` responses must not return `Content-Length`. 
> A server MUST NOT send a Content-Length header field in any response with a status code of [1xx (Informational)](https://httpwg.org/specs/rfc9110.html#status.1xx) or [204 (No Content)](https://httpwg.org/specs/rfc9110.html#status.204).